serverless/serverless
# Fix cleaning up stacks

## What did you implement:

***Implementing Issue:*** #2429

With this fix the now the last 4 stacks will be saved. A stack currently consists of 2 parts: the cloud formation template and the zip.

## How did you implement it:

I added two utility functions which group the s3 objects by the directory. This way it is also future proof in case we intend to add more files to a stack.

Note: I added a utils folder creating functions that don't touch `this.` I believe this is the way we should go in the future. Checkout the unit tests how much simpler they are for these util functions.

## How can we verify it:

1. Create a service
2. Run `serverless deploy` two times.
3. Confirm in your s3 bucket there are two directories.
4. Run `serverless deploy` 4 more times.
3. Confirm in your s3 bucket there are five directories.

***Is this ready for review?:*** YES
